radar_control/radar_control.py

```diff
@@ -699,91 +698,112 @@ def radar():
-                            if first_integration:
-                                decimation_scheme = None
-                                first_integration = False
 
-                            if TIME_PROFILE:
-                                time_after_sequence_metadata = datetime.utcnow()
-                                sequence_metadata_time = time_after_sequence_metadata - time_now
-                                rad_ctrl_print('Sequence Metadata time: {}'.format(sequence_metadata_time))
-
-                            # beam_phase_dict is slice_id : list of beamdirs, where beamdir = list
-                            # of antenna phase offsets for all antennas for that direction ordered
-                            # [0 ... main_antenna_count, 0 ... interferometer_antenna_count]
-
-                            # SEND ALL PULSES IN SEQUENCE.
-                            # If we have sent first sequence already and there is only one unique
-                            #  pulse in the averaging period, send all repeats until end of the
-                            #  averaging period.
-                            if first_sequence_out and aveperiod.one_pulse_only:
-                                for pulse_index, pulse_dict in \
-                                        enumerate(sequence_dict_list[sequence_index]):
-                                    data_to_driver(driverpacket, radar_control_to_driver,
-                                                   options.driver_to_radctrl_identity,
-                                                   pulse_dict['samples_array'], experiment.txctrfreq,
-                                                   experiment.rxctrfreq, experiment.txrate,
-                                                   experiment.rxrate,
-                                                   sequence.numberofreceivesamples,
-                                                   sequence.seqtime,
-                                                   pulse_dict['startofburst'], pulse_dict['endofburst'],
-                                                   pulse_dict['timing'], seqnum_start + num_sequences,
-                                                   repeat=True)
-                            else:
-                                for pulse_index, pulse_dict in \
-                                        enumerate(sequence_dict_list[sequence_index]):
+                            # on first sequence, we make the first set of samples.
+                            if sequence_index not in transmit_metadata_tracker:
+                                transmit_metadata_tracker[sequence_index] = {}
+                                transmit_metadata_tracker[sequence_index][num_sequences] = sequence.make_sequence(beam_iter, num_sequences)
+
+
+                            def send_pulses():
+                                for transmit_metadata in transmit_metadata_tracker[sequence_index][num_sequences]:
```

Comment:
I'd prefer a different variable name here - transmit_metadata is actually data per pulse in the sequence so something with pulse in the name would be my preference.